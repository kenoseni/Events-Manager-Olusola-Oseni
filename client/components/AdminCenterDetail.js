import React, {Component} from 'react';
import {link} from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import * as centerActions  from '.././actions/CenterActions'
import { eventCenters } from '.././reducers/';
import Navbar2 from './Navbar2';
import AdminCenterEvents from './AdminCenterEvents'

class AdminCenterDetail extends Component {
  constructor(props) {
    super(props);
  }

  render () {
    const { center } = this.props;
    return (
      <div>
        <Navbar2 />
        <main>
          <div className="container">
            <div className="row" id="info">
              <div className="col-xs-12 col-md-4">
                <div className="card w-40">  
                  {/*<!--Card image-->*/}
                  <div className="view overlay hm-white-slight">
                    <img src="./img2.jpg" className="img-fluid" alt={center.name} style={{width: '100%', height: '250px'}} />
                  </div>   
                  {/*<!--Card content-->*/}
                  <div className="card-body">
                    {/*<!--Title-->*/}
                    <h5 className="card-title font-weight-bold text-center">{(center.name === undefined ? null: center.name.toUpperCase())}</h5>
                    {/*<!--Text-->*/}
                    <p className="card-text"></p>
                  </div>     
                </div>
                {/*<!--/.Card-->*/}   
              </div>
              <div className="col-xs-12 col-md-8" id="paradise">
                <div className="card" >
                  <div className="card-header text-right">
                    Details 
                  </div>
                  <div className="card-body">
                    <h4 className="card-title">Description</h4>
                    <h5 className="card-text">{center.description}</h5>
                    <hr />
                    <h4 className="card-text">Facilities</h4>
                    <ul>
                      <li>{center.facilities}</li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div className="row" id="space">
              <div className="col-xs-12 col-md-4" id="space2">
                <div className="card w-40">  
                  {/*<!--Card image-->*/}
                  <div className="view overlay hm-white-slight">
                    <img src="./img1.jpg" className="img-fluid" alt={center.name} style={{width: '100%', height: '250px'}} />
                  </div>   
                </div>
                {/*<!--/.Card-->*/}
              </div>
              <AdminCenterEvents/>
            </div>
          </div>
        </main>
      </div>
    )
  }
}

export default AdminCenterDetail;
