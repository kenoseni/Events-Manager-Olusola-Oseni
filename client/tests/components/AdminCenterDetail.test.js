import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminCenterDetail from '../../components/AdminCenterDetail';

configure({ adapter: new Adapter() });
const mocksearchForCenters = jest.fn()

let props = {
  center: {},
  id: 1,
  center: {
      events: []
  }
}

const wrapper = shallow(<AdminCenterDetail
  searchForCenters={mocksearchForCenters}
  {...props}
/>)

describe('<AdminCenterDetails />', () => {
  it('should render a <AdminCenterDetail /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
