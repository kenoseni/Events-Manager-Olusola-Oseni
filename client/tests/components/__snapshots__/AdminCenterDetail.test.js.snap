// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminCenterDetails /> should render a <AdminCenterDetail /> component 1`] = `
<div>
  <Navbar
    centerDetails="CenterDetails"
    centers="Centers"
    events="Events"
    home="Home"
    id={1}
    page="CenterDetails"
    searchForCenters={[MockFunction]}
  />
  <main>
    <div
      className="container"
    >
      <div
        className="row"
        id="info"
      >
        <div
          className="col-xs-12 col-md-4"
        >
          <div
            className="card w-40"
          >
            <div
              className="view overlay hm-white-slight"
            >
              <img
                className="img-fluid"
                style={
                  Object {
                    "height": "250px",
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              className="card-body"
            >
              <h5
                className="card-title font-weight-bold text-center"
              />
              <p
                className="card-text"
              />
            </div>
          </div>
        </div>
        <div
          className="col-xs-12 col-md-8"
          id="paradise"
        >
          <div
            className="card"
          >
            <div
              className="card-header text-right"
            >
              Details
            </div>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Description
              </h5>
              <h6
                className="card-text"
              />
              <hr />
              <h5
                className="card-title"
              >
                Location
              </h5>
              <h6
                className="card-text"
              />
              <hr />
              <h5
                className="card-title"
              >
                Price
              </h5>
              <h6
                className="card-text"
              />
              <hr />
              <h5
                className="card-title"
              >
                Capacity
              </h5>
              <h6
                className="card-text"
              >
                 guests
              </h6>
              <hr />
              <h5
                className="card-title"
              >
                Facilities
              </h5>
              <h6
                className="card-text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
        id="space"
      >
        <div
          className="col-xs-12 col-md-4"
          id="space2"
        />
        <AdminCenterEvents
          events={Array []}
        />
      </div>
    </div>
  </main>
</div>
`;
